### PR TITLE
Security config dev branch

### PR DESCRIPTION
**Pull request by SebastianBoe**
see commit messages

### CODE REVIEW

**Review by vili-nordic — APPROVED** [review]

**Comment by SebastianBoe** [comment]
@vidar: Added some more unrelated commits FYI, so this is now a dev branch (but I still want review).

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c [hunk 1]
```diff
@@ -1289,15 +1289,48 @@ static const uint32_t target_peripherals[] = {
...
+#endif
+#ifdef TFM_PERIPHERAL_GPIO2_PIN_MASK_SECURE
+		TFM_PERIPHERAL_GPIO2_PIN_MASK_SECURE,
 #endif
+	};
+
+	/* Note that there are two different API's for SPU configuration */
+#if NRF_SPU_HAS_MEMORY
+
+	for(int port = 0; port < ARRAY_SIZE(secure_pins); port++){
```
Comment (vili-nordic) [inline platform/ext/target/nordic_nrf/common/core/target_cfg.c:1305]: Rename variable port -> pin ? Maybe?

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c [hunk 2]
```diff
@@ -1289,15 +1289,48 @@ static const uint8_t target_peripherals[] = {
...
+		nrf_spu_gpio_config_set(NRF_SPU, port, secure_pins[port], SPU_LOCK_CONF_LOCKED);
+	}
 
+#elif NRF_SPU_HAS_PERIPHERAL_ACCESS
+
+	for(int port = 0; port < 2; port++) {
```
Comment (vili-nordic) [inline platform/ext/target/nordic_nrf/common/core/target_cfg.c:1311]: Is hard coding this to 2 right, or should it be ARRAY_SIZE(secure_pins) like in the previous case?

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c (hunk 1 above)
Comment (SebastianBoe) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:1305]: hmm, If you have misread then I am sure I can improve somehow.

But here, port is 0 or 1 and represents P0 or P1.

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c (hunk 2 above)
Comment (SebastianBoe) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:1311]: I can do ARRAY_SIZE :)

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c (hunk 1 above)
Comment (vili-nordic) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:1305]: hmm. Ok, I see that now

**Comment by SebastianBoe** [comment]
@vili-nordic : Pushed 4 more commits ontill top.

still waiting for a review from someone else...

@mswarowsky , @Vge0rge

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c (hunk 1 above)
Comment (mswarowsky) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:1275]: It confuses a bit at first look that the array is called xx_pins and then the iteration variable is called port

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c [hunk 3]
```diff
@@ -1083,6 +1083,26 @@ enum tfm_plat_err_t spu_periph_init_cfg(void)
...
+	/* Configure the CTRL-AP mailbox interface to be secure as it is used by the secure ADAC service */
+	spu_peripheral_config_secure(NRF_CTRLAP_S_BASE, SPU_LOCK_CONF_LOCKED);
+
+	/* Configure NRF_MEMCONF to be secure as it could otherwise be used to corrupt secure RAM. */
+	spu_peripheral_config_secure(NRF_MEMCONF_S_BASE, SPU_LOCK_CONF_LOCKED);
+
+	/* Configure trace to be secure, as the security implications of non-secure trace are not understood */
+	spu_peripheral_config_secure(NRF_TAD_S_BASE, SPU_LOCK_CONF_LOCKED);
+
+	/* Configure these secret HW features to be secure, as the security implications of them being non-secure are not understood */
```
Comment (mswarowsky) [inline platform/ext/target/nordic_nrf/common/core/target_cfg.c:1095]: Didn't you mean to  say `That these addresses are just  missing in MDK therefore using the addressed hard coded to configure them`

**Review by mswarowsky — APPROVED** [review]
Maybe add 
fixup! [nrf noup] platform: nordic_nrf: Add support for 54l
to all the noup's, else just a small nit, but not a blocker

**Pull request merged**